gtsam/geometry/SOn.h

```diff
@@ -0,0 +1,331 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010-2019, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    SOn.h
+ * @brief   n*n matrix representation of SO(n), template on N, which can be
+ * Eigen::Dynamic
+ * @author  Frank Dellaert
+ * @date    March 2019
+ */
+
+#pragma once
+
+#include <gtsam/base/Lie.h>
+#include <gtsam/base/Manifold.h>
+
+#include <Eigen/Core>
+#include <boost/random.hpp>
+
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace gtsam {
+
+namespace internal {
+/// Calculate dimensionality of SO<N> manifold, or return Dynamic if so
+constexpr int DimensionSO(int N) {
+  return (N < 0) ? Eigen::Dynamic : N * (N - 1) / 2;
+}
+
+// Calculate N^2 at compile time, or return Dynamic if so
+constexpr int NSquaredSO(int N) { return (N < 0) ? Eigen::Dynamic : N * N; }
+}  // namespace internal
+
+/**
+ * Manifold of special orthogonal rotation matrices SO<N>.
+ * Template paramater N can be a fizxed integer or can be Eigen::Dynamic
+ */
+template <int N>
+class SO : public LieGroup<SO<N>, internal::DimensionSO(N)> {
+ public:
+  enum { dimension = internal::DimensionSO(N) };
```

Comment:
You might want to add `EIGEN_MAKE_ALIGNED_OPERATOR_NEW` macro here.

Maybe that's what is causing the segfaults.

Reply:
Thank you for the suggestion! Testing now.

Reply:
It still segfaults, so probably need further investigation.

Reply:
Try disabling the "Build for native architecture" flag in cmake (using ccmake or cmake-gui)